src/middleware/cors.rs

```diff
@@ -395,7 +398,7 @@ mod test {
     #[test]
     fn not_set_origin_header() {
         let mut app = app();
-        app.middleware(Cors::new());
+        app.middleware(Cors::new().allow_origin(ALLOW_ORIGIN));
```

Comment:
The test succeeded before because the default setting is `Origin::Any` which is successful for empty origin header values (as described in the PR message). Adding a specific header was necessary to make the test fail (before this PR).